src/grt/src/GlobalRouter.cpp

```diff
@@ -1659,6 +1659,22 @@ bool GlobalRouter::hasAvailableResources(bool is_horizontal,
   return cap > 0;
 }
 
+// Find the position of the middle of a GCell closest to the position
+odb::Point GlobalRouter::getPositionOnGrid(const odb::Point& real_position)
+{
+  // Get x grid position
+  int grid_x
+      = (int) ((real_position.x() - grid_->getXMin()) / grid_->getTileSize());
+  // Get x real position
+  int new_x = grid_->getTileSize() * (grid_x + 0.5) + grid_->getXMin();
+  // Get y grid position
+  int grid_y
+      = (int) ((real_position.y() - grid_->getYMin()) / grid_->getTileSize());
+  // Get y real position
+  int new_y = grid_->getTileSize() * (grid_y + 0.5) + grid_->getYMin();
+  return odb::Point(new_x, new_y);
+}
```

Comment:
You are duplicating code here. Instead of implementing a new function, you can call `Grid::getPositionOnGrid`. The `grid_` object should be initialized at the time you call it.